concordium-std/src/impls.rs

```diff
@@ -145,10 +146,10 @@ impl HasContractStateEntry for StateEntry {
     }
 
     fn get_key(&self) -> Vec<u8> {
```

Comment:
I don't think this method should return a `Vec<u8>`. The principle we have until now is that it is in principle possible to go without an allocator. So our API should as much as practical not rely on an allocator being present.

Instead the buffer should be provided by the caller.